plugins/org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/resourceFactory/ResourceFactoryFragment2.xtend

```diff
@@ -0,0 +1,64 @@
+/*******************************************************************************
+ * Copyright (c) 2015 itemis AG (http://www.itemis.eu) and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *******************************************************************************/
+package org.eclipse.xtext.xtext.generator.resourceFactory
+
+import org.eclipse.xtext.xtext.generator.AbstractGeneratorFragment2
+import com.google.inject.Inject
+import org.eclipse.xtext.xtext.generator.XtextGeneratorNaming
+import org.eclipse.xtext.resource.IResourceFactory
+import org.eclipse.emf.ecore.resource.Resource
+import org.eclipse.xtext.resource.IResourceServiceProvider
+
+/**
+ * Contributes the registration <code>org.eclipse.emf.ecore.extension_parser</code>
+ * to the Eclipse (UI) plugin's plugin.xml. See
+ * <a href="https://bugs.eclipse.org/bugs/show_bug.cgi?id=264578">
+ * https://bugs.eclipse.org/bugs/show_bug.cgi?id=264578</a> for details.
+ * 
+ * @author Christian Schneider - Initial contribution and API
+ */
+class ResourceFactoryFragment2 extends AbstractGeneratorFragment2 {
+	
+	@Inject
+	extension XtextGeneratorNaming
+	
+	override generate() {
+		
+		language.runtimeGenSetup.registrations.add('''
+			�IResourceFactory� resourceFactory = injector.getInstance(IResourceFactory.class);
```

Comment:
The second occurrence of IResourceFactory should be a type literal, too, e.g. `«IResourceFactory»`
Same in the line below.